### PR TITLE
Update Solana SDK to v1.17

### PR DESCRIPTION
**Pull request by anton-lisanin**
Update Solana SDK to 1.17.23 and Rust to 1.73

### CODE REVIEW

**Comment by neonlabstech** [comment]
<details><summary>Dapps report</summary>


Cost report for "Aave" dApp

| Action | Fee | Cost in $ | Accounts | TRx | Estimated Gas | Used Gas | Used % of EG |
| --- | --- | --- | --- | --- | --- | --- |--- |
| Token mint | 0.00233072 | 0.002772631154552 | 9 | 1 | 1185360 | 1165360 | 98.31 | 
| Token approve | 0.00233072 | 0.002772631154552 | 9 | 1 | 1185360 | 1165360 | 98.31 | 
| Deposit to lending pool | 0.00791088 | 0.009410805394008001 | 4 | 5 | 4824480 | 3955440 | 81.99 | 
| Borrow from lending pool | 0.00517144 | 0.0061520464359384 | 4 | 30 | 3779120 | 2585720 | 68.42 | 
| Repay | 0.00049 | 0.0005829111589 | 4 | 27 | 2586800 | 245000 | 9.47 | 
| Flashloan | 0.00510144 | 0.006068687563104 | 4 | 28 | 7464880 | 2550720 | 34.17 | 
| Withdraw | 7e-05 | 8.327219949999999e-05 | 4 | 5 | 1857760 | 35000 | 1.88 | 
| Liquidation | 0.00340072 | 0.004045506204052 | 4 | 57 | 4960880 | 1700360 | 34.28 | 

Cost report for "Uniswap V3" dApp

| Action | Fee | Cost in $ | Accounts | TRx | Estimated Gas | Used Gas | Used % of EG |
| --- | --- | --- | --- | --- | --- | --- |--- |
| Token approve | 0.00233072 | 0.002772630688408 | 9 | 1 | 1185360 | 1165360 | 98.31 | 
| NonfungiblePositionManager - Mint position | 0.02772144 | 0.032977498485816 | 4 | 28 | 31261120 | 13860720 | 44.34 | 
| NonfungiblePositionManager - Increase liquidity  | 0.00044 | 0.0005234251660000001 | 23 | 23 | 4364240 | 220000 | 5.04 | 
| NonfungiblePositionManager - Decrease Liquidity | 0.00139808 | 0.001663159672912 | 17 | 1 | 4274240 | 699040 | 16.35 | 
| NonfungiblePositionManager - Collect Fees | 0.00464144 | 0.005521469323816 | 20 | 1 | 3539120 | 2320720 | 65.57 | 
| NonfungiblePositionManager - Burn Liquidity Position | 2e-05 | 2.3792017000000002e-05 | 15 | 1 | 3954560 | 10000 | 0.25 | 
| Direct swap | 2e-05 | 2.3792017000000002e-05 | 17 | 1 | 1328400 | 10000 | 0.75 | 
| Burn transaction | 0.00047936 | 0.000570247063456 | 11 | 1 | 739040 | 239680 | 32.43 | 
| Collect transaction | 0.00233072 | 0.002772626493112 | 12 | 1 | 1654720 | 1165360 | 70.43 | 

Cost report for "Saddle Finance" dApp

| Action | Fee | Cost in $ | Accounts | TRx | Estimated Gas | Used Gas | Used % of EG |
| --- | --- | --- | --- | --- | --- | --- |--- |
| Add liquidity | 0.00234072 | 0.002784522501612 | 3 | 2 | 3202800 | 1170360 | 36.54 | 
| Swap DAI -> USDC | 2e-05 | 2.3792017000000002e-05 | 21 | 1 | 1797760 | 10000 | 0.56 | 
| Swap USDC -> DAI | 2e-05 | 2.3792017000000002e-05 | 21 | 1 | 1797760 | 10000 | 0.56 | 
| Remove liquidity | 2e-05 | 2.3792017000000002e-05 | 23 | 1 | 1937440 | 10000 | 0.52 | 
| Add liquidity in 3 Tokens | 0.00283072 | 0.0033674089430400004 | 4 | 30 | 4241520 | 1415360 | 33.37 | 
| Swap DAI -> USDC 3 pool | 2e-05 | 2.3791890000000004e-05 | 21 | 1 | 1807760 | 10000 | 0.55 | 
| Swap USDC -> USDT 3 pool | 2e-05 | 2.3791890000000004e-05 | 21 | 1 | 1807760 | 10000 | 0.55 | 
| Swap USDC -> DAI 3 pool | 2e-05 | 2.3791890000000004e-05 | 21 | 1 | 1807760 | 10000 | 0.55 | 
| Remove liquidity 3 pool | 3e-05 | 3.56722668e-05 | 3 | 2 | 2646480 | 15000 | 0.57 | 
| Add liquidity in metapool | 0.00290072 | 0.0034491752584032 | 3 | 31 | 3292800 | 1450360 | 44.05 | 
| Swap SUSD -> LP metapool | 2e-05 | 2.3781511200000003e-05 | 22 | 1 | 1807760 | 10000 | 0.55 | 
| Swap USDC -> SUSD metapool | 0.00103 | 0.0012247007249 | 4 | 53 | 3775520 | 515000 | 13.64 | 
| Swap DAI -> USDT metapool | 0.00056 | 0.0006658707048 | 4 | 29 | 3086160 | 280000 | 9.07 | 
| Remove liquidity in Metapool | 2e-05 | 2.37776248e-05 | 23 | 1 | 1937440 | 10000 | 0.52 |

**Comment by neonlabstech** [comment]
<details><summary>Dapps report</summary>

File: evm_loader/program/src/executor/precompile_extension/metaplex.rs [hunk 1]
```diff
@@ -161,37 +195,36 @@ impl<B: AccountStorage> ExecutorState<'_, B> {
...
-        );
+        let (metadata_pubkey, _) = Metadata::find_pda(&mint);
+
+        let instruction = CreateMetadataAccountV3Builder::new()
+            .metadata(metadata_pubkey)
+            .mint(mint)
+            .mint_authority(signer_pubkey)
+            .update_authority(signer_pubkey, true)
+            .payer(self.backend.operator())
+            .is_mutable(true)
```
Comment (andreisilviudragnea) [inline evm_loader/program/src/executor/precompile_extension/metaplex.rs:206]: This was `false` before. Is this expected?

**Comment by neonlabstech** [comment]
<details><summary>Dapps report</summary>


Cost report for "Aave" dApp

| Action | Fee | Cost in $ | Accounts | TRx | Estimated Gas | Used Gas | Used % of EG |
| --- | --- | --- | --- | --- | --- | --- |--- |
| Token mint | 0.00233072 | 0.0028867210872192003 | 9 | 1 | 1185360 | 1165360 | 98.31 | 
| Token approve | 0.00233072 | 0.0028867210872192003 | 9 | 1 | 1185360 | 1165360 | 98.31 | 
| Deposit to lending pool | 0.00791088 | 0.0097980470045568 | 4 | 5 | 4824480 | 3955440 | 81.99 | 
| Borrow from lending pool | 0.00517144 | 0.006405104388038401 | 4 | 30 | 3779120 | 2585720 | 68.42 | 
| Repay | 0.00049 | 0.0006068911464 | 4 | 27 | 2586800 | 245000 | 9.47 | 
| Flashloan | 0.00510144 | 0.0063184056528384 | 4 | 28 | 7464880 | 2550720 | 34.17 | 
| Withdraw | 7e-05 | 8.669873519999999e-05 | 4 | 5 | 1857760 | 35000 | 1.88 | 
| Liquidation | 0.00340072 | 0.0042119731824192 | 4 | 57 | 4501520 | 1700360 | 37.77 | 

Cost report for "Uniswap V3" dApp

| Action | Fee | Cost in $ | Accounts | TRx | Estimated Gas | Used Gas | Used % of EG |
| --- | --- | --- | --- | --- | --- | --- |--- |
| Token approve | 0.00233072 | 0.0028867210872192003 | 9 | 1 | 1185360 | 1165360 | 98.31 | 
| NonfungiblePositionManager - Mint position | 0.02772144 | 0.034334482656038404 | 4 | 28 | 31261120 | 13860720 | 44.34 | 
| NonfungiblePositionManager - Increase liquidity  | 0.00044 | 0.0005449634784 | 23 | 23 | 4364240 | 220000 | 5.04 | 
| NonfungiblePositionManager - Decrease Liquidity | 0.00139808 | 0.0017313599446623998 | 17 | 1 | 4274240 | 699040 | 16.35 | 
| NonfungiblePositionManager - Collect Fees | 0.00464144 | 0.0057478851722032 | 20 | 1 | 3539120 | 2320720 | 65.57 | 
| NonfungiblePositionManager - Burn Liquidity Position | 2e-05 | 2.4771067200000004e-05 | 15 | 1 | 3954560 | 10000 | 0.25 | 
| Direct swap | 2e-05 | 2.4771067200000004e-05 | 17 | 1 | 1328400 | 10000 | 0.75 | 
| Burn transaction | 0.00047936 | 0.0005937129386496 | 11 | 1 | 739040 | 239680 | 32.43 | 
| Collect transaction | 0.00233072 | 0.0028867210872192003 | 12 | 1 | 1654720 | 1165360 | 70.43 | 

Cost report for "Saddle Finance" dApp

| Action | Fee | Cost in $ | Accounts | TRx | Estimated Gas | Used Gas | Used % of EG |
| --- | --- | --- | --- | --- | --- | --- |--- |
| Add liquidity | 0.00234072 | 0.0028991066208192 | 3 | 2 | 3202800 | 1170360 | 36.54 | 
| Swap DAI -> USDC | 2e-05 | 2.4771067200000004e-05 | 21 | 1 | 1797760 | 10000 | 0.56 | 
| Swap USDC -> DAI | 2e-05 | 2.4771067200000004e-05 | 21 | 1 | 1797760 | 10000 | 0.56 | 
| Remove liquidity | 2e-05 | 2.4771067200000004e-05 | 23 | 1 | 1937440 | 10000 | 0.52 | 
| Add liquidity in 3 Tokens | 0.00283072 | 0.0035059977672192003 | 4 | 30 | 4241520 | 1415360 | 33.37 | 
| Swap DAI -> USDC 3 pool | 2e-05 | 2.4771067200000004e-05 | 21 | 1 | 1807760 | 10000 | 0.55 | 
| Swap USDC -> USDT 3 pool | 2e-05 | 2.4771067200000004e-05 | 21 | 1 | 1807760 | 10000 | 0.55 | 
| Swap USDC -> DAI 3 pool | 2e-05 | 2.4771067200000004e-05 | 21 | 1 | 1807760 | 10000 | 0.55 | 
| Remove liquidity 3 pool | 3e-05 | 3.71566008e-05 | 3 | 2 | 2646480 | 15000 | 0.57 | 
| Add liquidity in metapool | 0.00290072 | 0.0035926965024192 | 3 | 31 | 3292800 | 1450360 | 44.05 | 
| Swap SUSD -> LP metapool | 2e-05 | 2.4771067200000004e-05 | 22 | 1 | 1807760 | 10000 | 0.55 | 
| Swap USDC -> SUSD metapool | 0.00103 | 0.0012757099608000002 | 4 | 53 | 3775520 | 515000 | 13.64 | 
| Swap DAI -> USDT metapool | 0.00056 | 0.0006935898815999999 | 4 | 30 | 3086160 | 280000 | 9.07 | 
| Remove liquidity in Metapool | 2e-05 | 2.4771067200000004e-05 | 23 | 1 | 1937440 | 10000 | 0.52 |

**Review by andreisilviudragnea — COMMENTED** [review]
Should we also replace `cargo build-bpf` with `cargo build-sbf`?

File: evm_loader/program/src/executor/precompile_extension/metaplex.rs (hunk 1 above)
Comment (anton-lisanin) [reply evm_loader/program/src/executor/precompile_extension/metaplex.rs:206]: Yes. This change requested by @s-medvedev

**Review by andreisilviudragnea — APPROVED** [review]

**Pull request merged**